Source/RESTyard.Client/Reader/SirenHypermediaReader.cs

```diff
@@ -37,94 +40,137 @@ public class SirenHypermediaReader : IHypermediaReader
+                }
+                var result = this.ReadHypermediaObject(rootObject, resolver);
+                return result.Bind<(HypermediaClientObject, string)>(
+                    ok => (ok, rootObject.Serialize()));
+            }
+            catch (Exception e)
+            {
+                return HypermediaReaderResult.Error<(HypermediaClientObject, string)>(HypermediaReaderProblem.Exception(e));
+            }
         }
 
-        private HypermediaClientObject ReadHypermediaObject(
+        private HypermediaReaderResult<HypermediaClientObject> ReadHypermediaObject(
             IToken rootObject,
             IHypermediaResolver resolver)
         {
-            var classes = ReadClasses(rootObject);
-            var hypermediaObjectInstance = this.hypermediaObjectRegister.CreateFromClasses(classes);
-
-            this.ReadTitle(hypermediaObjectInstance, rootObject);
-            this.ReadRelations(hypermediaObjectInstance, rootObject);
-            this.FillHypermediaProperties(hypermediaObjectInstance, rootObject, resolver);
-            
-
-            return hypermediaObjectInstance;
+            return ReadClasses(rootObject)
+                .Bind(classes =>
+                {
+                    return this.hypermediaObjectRegister.CreateFromClasses(classes)
+                        .Match(
+                            hypermediaObjectInstance =>
+                            {
+                                this.ReadTitle(hypermediaObjectInstance, rootObject);
+                                this.ReadRelations(hypermediaObjectInstance, rootObject);
+                                return this.FillHypermediaProperties(hypermediaObjectInstance, rootObject, resolver)
+                                    .Map(_ => hypermediaObjectInstance);
+                            },
+                            error => HypermediaReaderResult.Error<HypermediaClientObject>(
+                                HypermediaReaderProblem.InvalidClientClass(error)));
+
+                });
         }
 
-        private void FillHypermediaProperties(
+        private HypermediaReaderResult<Unit> FillHypermediaProperties(
             HypermediaClientObject hypermediaObjectInstance,
             IToken rootObject,
             IHypermediaResolver resolver)
         {
             var typeInfo = hypermediaObjectInstance.GetType().GetTypeInfo();
             var properties = typeInfo.GetProperties(BindingFlags.Public | BindingFlags.Instance);
-            foreach (var propertyInfo in properties)
+            HypermediaReaderResult<Unit> result = HypermediaReaderResult.Ok(No.Thing);
+            bool HasSetter(PropertyInfo p) => p.CanWrite;
+            bool IsIgnored(PropertyInfo p) =>
+                p.GetCustomAttribute<ClientIgnoreHypermediaPropertyAttribute>() is not null;
+            foreach (var propertyInfo in properties.Where(p => HasSetter(p) && !IsIgnored(p)))
```

Comment:
Explicit Focus: Change not in description
If the HCO has Get-Only properties, e.g. to map a collection of entities to the single entity that is expected, the reader will return a NullReferenceException Error. Instead just filter those properties that can actually be written.